extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryProducer.java

```diff
@@ -154,10 +155,20 @@ private Map<String, String> getOtelConfigs() {
                 if (configValue.getValue() != null) {
                     NameIterator name = new NameIterator(propertyName);
                     name.next();
-                    oTelConfigs.put(name.getName().substring(name.getPosition() + 1), configValue.getValue());
+                    String propertyValue = getConfigPropertyValue(propertyName, configValue);
+                    oTelConfigs.put(name.getName().substring(name.getPosition() + 1), propertyValue);
                 }
             }
         }
         return oTelConfigs;
     }
+
+    private static String getConfigPropertyValue(String propertyName, ConfigValue smallRyeConfig) {
```

Comment:
I think you can move this static method to util as well. It will contain the needed logic in one spot.
Calling it something like normalizeDuration would also help.